### PR TITLE
【增加】stm32f103-dofly-M3S开发板的BSP

### PR DESCRIPTION
**Pull request by willianchanlovegithub**
Signed-off-by: Willian Chan <chentingwei@rt-thread.com>

## 拉取/合并请求描述：(PR description)

[
增加 stm32f103-dofly-M3S 开发板的BSP
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by BernardXiong** [comment]
请签署CLA

File: bsp/stm32/stm32f103-dofly-M3S/board/Kconfig [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                depends on BSP_USING_UART1 && RT_SERIAL_USING_DMA
+                default n
+        endif
+
+    menuconfig BSP_USING_SPI
+        bool "Enable SPI BUS"
+        default n
+        select RT_USING_SPI
+        if BSP_USING_SPI
+            config BSP_USING_SPI1
```
Comment (tyustli) [inline bsp/stm32/stm32f103-dofly-M3S/board/Kconfig]: 这里支持的情况请和README文件保持一致

File: bsp/stm32/stm32f103-dofly-M3S/board/Kconfig [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                default n
+                
+            config BSP_SPI1_RX_USING_DMA
+                bool "Enable SPI1 RX DMA"
+                depends on BSP_USING_SPI1
+                select BSP_SPI1_TX_USING_DMA
+                default n
+        endif
+
+    menuconfig BSP_USING_I2C1
```
Comment (tyustli) [inline bsp/stm32/stm32f103-dofly-M3S/board/Kconfig]: 这里支持的情况请和README文件保持一致

File: bsp/stm32/stm32f103-dofly-M3S/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+```bash
+ \ | /
+- RT -     Thread Operating System
+ / | \     4.0.1 build Apr  9 2019
+ 2006 - 2019 Copyright by rt-thread team
+msh >
+```
+### 进阶使用
+
+此 BSP 默认只开启了 GPIO 和 串口1 的功能，如果需使用 SD 卡、Flash 等更多高级功能，需要利用 ENV 工具对BSP 进行配置，步骤如下：
```
Comment (tyustli) [inline bsp/stm32/stm32f103-dofly-M3S/README.md]: 这里串口是1，但是前面初始化的是串口3，请问是不是冲突

File: bsp/stm32/stm32f103-dofly-M3S/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+## 注意事项
+
+- 这款开发板使用终端工具 SecureCRT 来打开串口会出现系统不能启动的问题，推荐使用 XCOM V2.0。
+
+## 联系人信息
+
+维护人:
+
+-  [WillianChan](https://github.com/willianchanlovegithub), 邮箱：<willianchan_lovelife@foxmail.com>
```
Comment (tyustli) [inline bsp/stm32/stm32f103-dofly-M3S/README.md]: 这里请把个人邮箱删掉

**Pull request merged**